File: find_min_rotated.go
package binary_search

func findMin(nums []int) int {
	var (
		l = 0
		r = len(nums)
		m int
	)
	for l < r {
		m = l + (r-1)/2
		if nums[l] < nums[r] {
			r = m
		} else {
			l = m + 1
		}
	}
	return nums[l]
}

File: find_min_rotated.py
# Can you write an algorithm that runs in O(log(n)) time?
class Solution:
    def findMin(self, nums: list[int]) -> int:
        left, right = 0, len(nums) - 1
        while left < right:
            mid = left + (right - 1) // 2
            if nums[mid] < nums[right]:
                right = mid
            else:
                left = mid + 1
        return nums[left]
